File: Data Structures & Algorithms/eating-bananas/submission-4.cpp
class Solution {
public:
    bool check(long long mid, int h, vector<int> &piles) {
        long long sum = 0;
        for(int i : piles) {
            sum += (i + mid - 1) / mid;
        }
        return sum <= h;
    }
    int minEatingSpeed(vector<int>& piles, int h) {
        int n = piles.size();
        long long lo = 1, hi = 1e10, ans = *max_element(piles.begin(), piles.end());
        while(lo <= hi) {
            long long mid = (lo + hi) / 2;
            if(check(mid, h, piles)) {
                hi = mid - 1;
                ans = mid;
            } else {
                lo = mid + 1;
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/count-number-of-islands/submission-4.cs
public class Solution {
    int[] dx = {1, 0, -1, 0};
    int[] dy = {0, 1, 0, -1};
    void dfs(int r, int c, char[][] grid, int n, int m) {
        if (r < 0 || c < 0 || r >= n || c >= m || grid[r][c] == '0')
            return;
        grid[r][c] = '0';
        for (int i = 0; i < 4; i++) {
            dfs(r + dx[i], c + dy[i], grid, n, m);
        }
    }
    public int NumIslands(char[][] grid) {
        if (grid.Length == 0) return 0;
        int n = grid.Length;
        int m = grid[0].Length;
        int ans = 0;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (grid[i][j] == '1') {
                    dfs(i, j, grid, n, m);
                    ans++;
                }
            }
        }
        return ans;
    }
}